src/main/java/org/gridsuite/study/server/service/NetworkModificationTreeService.java

```diff
@@ -932,4 +932,37 @@ public long countBuiltNodes(UUID studyUuid, UUID rootNetworkUuid) {
         // perform N queries, but it's fast: 25 ms for 400 nodes
         return nodes.stream().filter(n -> self.getNodeBuildStatus(n.getIdNode(), rootNetworkUuid).isBuilt()).count();
     }
+
+    @Transactional
+    public List<NodeAlias> getNodeAliases(UUID nodeId) {
+        NetworkModificationNodeInfoEntity node = getNetworkModificationNodeInfoEntity(nodeId);
+        Map<NodeType, List<UUID>> nodeUuidsByType = node.getNodeAliases().stream().map(NodeAliasEmbeddable::getReferencedNode)
+            .collect(Collectors.groupingBy(NodeEntity::getType, Collectors.mapping(NodeEntity::getIdNode, Collectors.toList())));
+
+        Map<UUID, String> nodeNames = new HashMap<>();
+        if (nodeUuidsByType.get(NodeType.NETWORK_MODIFICATION) != null) {
+            nodeNames.putAll(networkModificationNodeInfoRepository.findAllById(nodeUuidsByType.get(NodeType.NETWORK_MODIFICATION))
+                .stream().collect(Collectors.toMap(AbstractNodeInfoEntity::getIdNode, AbstractNodeInfoEntity::getName)));
+        }
+        if (nodeUuidsByType.get(NodeType.ROOT) != null) {
+            nodeNames.putAll(rootNodeInfoRepository.findAllById(nodeUuidsByType.get(NodeType.ROOT))
+                .stream().collect(Collectors.toMap(AbstractNodeInfoEntity::getIdNode, AbstractNodeInfoEntity::getName)));
+        }
+
+        return node.getNodeAliases().stream().map(nodeAlias ->
+            nodeAlias.toNodeAlias(nodeNames.get(nodeAlias.getReferencedNode().getIdNode()))).toList();
+    }
+
+    @Transactional
+    public void updateNodeAliases(UUID nodeId, List<NodeAlias> nodeAliases) {
+        NetworkModificationNodeInfoEntity node = getNetworkModificationNodeInfoEntity(nodeId);
```

Comment:
this throws an exception if nodeId is the root node ...

Reply:
Same as above